qmulticast/utils/multipartite_network.py

```diff
@@ -205,3 +139,67 @@ def create_multipartite_network(name: str, graph: DiGraph) -> Network:
             mem_position += 2
 
     return network
+
+
+def add_mulitpartite_components(node,node_name,port_size,source_delay,source_noise,state_sampler):
```

Comment:
give type hints for each of the inputs. You also don't need to pass `node_name` and `port_size` since you already send in that data with `node`